.github/workflows/prototype-tests.yml

```
@@ -40,13 +40,18 @@ jobs:
         run: pip install --progress-bar=off pytest pytest-mock
 
       - name: Run prototype datasets tests
+        id: datasets
         shell: bash
         run: pytest --durations=20 test/test_prototype_datasets*.py
 
       - name: Run prototype transforms tests
+        id: transforms
+        if: ${{ success() || steps.datasets.conclusion == 'failure' }}
         shell: bash
         run: pytest --durations=20 test/test_prototype_transforms*.py
 
       - name: Run prototype models tests
+        if: ${{ success() || steps.datasets.conclusion == 'failure' || steps.transforms.conclusion == 'failure' }}
```

> They are not redundant. If we leave them out, we are left with if: success() which is the default behavior if you don't define if: at all. Meaning, the step will only be run if all previous steps have succeeded.

I don't think @YosuaMichael  is suggesting to remove all `XYZ == 'failure'` checks. From what I understand I think he's suggesting the  following change only at this specific line (which I agree seems like it should work):

 ```suggestion
         if: ${{ success() || steps.transforms.conclusion == 'failure' }}
```

---

Although I guess that

`success() || steps.transforms.conclusion == 'failure'`

will evaluate to False if `transforms` passed but `datasets` failed...? In which case I agree we need the long form.

---

What if `steps.datasets.conclusion == 'failure'` and `steps.transforms.conclusion == 'success'`? `success()` will be `false` since we have at least one failure. 

In this case `success() || steps.transforms.conclusion == 'failure'` boils down to `false || false` -> `false` and the step is not run. 

With my version we get `false || true || false` -> `true` and the step will be run.

In general we need an expression that makes sure that the setup has succeeded. This is achieved by checking `success()` or failure of any of the steps afterwards. 

Branching is a pain in CI in general, but I think with these three conditions it is somewhat tolerable. Given that this sparked quite a bit of confusion maybe we can refactor this a little. I see two options there:

1. Change the condition to `if: steps.$LAST_SETUP_STEP.conclusion == 'success'`.
2. Add a another dummy step after the setup that we can check for success

From these two I like 2. better since then we only need to remember to keep it as a last step of the setup but can do whatever we want before it.

---

I've implemented your suggestion in 89f83067817f454bb5a7b919d38f8661ca23de20: https://github.com/pytorch/vision/actions/runs/3112539248/jobs/5046070450 prototype tests are skipped although they should be run.

---

I went for 2. in 7a9eb1a00a35a19ff82606c8130222cf61d9cdbe and it works out well. There is an outage on the datasets tests, but the others are still run

---

Sorry, was otw to the office just now. Yeah, what I meant is like what @NicolasHug suggest, and from @pmeier explanation I can understand why, thanks for that.

I also saw your changes now using:
```
success() || ( failure() && steps.setup.conclusion == 'success' )
```

I think this is good (and avoid chaining)! Thanks for this!